pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -282,6 +293,121 @@ func configMapDeleted(obj interface{}) {
 	}
 }
 
+// initVolumeHandleToPvcMap performs all the operations required to initialize the volume id to PVC name map.
+// It also watches for PV update & delete operations, and updates the map accordingly.
+func initVolumeHandleToPvcMap(ctx context.Context, k8sClient clientset.Interface) {
+	log := logger.GetLogger(ctx)
+	log.Debugf("Initializing volume ID to PVC name map")
+	k8sOrchestratorInstance.volumeIDToPvcMap = make(map[string]string)
+
+	// Set up kubernetes resource listener to listen events on PersistentVolume
+	k8sOrchestratorInstance.informerManager = k8s.NewInformer(k8sClient)
+	k8sOrchestratorInstance.informerManager.AddPVListener(
+		func(obj interface{}) { // Add
+			pvAdded(obj)
+		},
+		func(oldObj interface{}, newObj interface{}) { // Update
+			pvUpdated(oldObj, newObj)
+		},
+		func(obj interface{}) { // Delete
+			pvDeleted(obj)
+		})
+
+	k8sOrchestratorInstance.informerManager.AddPVCListener(
+		func(obj interface{}) { // Add
+			pvcAdded(obj)
+		},
+		nil, // Update
+		nil, // Delete
+	)
+	k8sOrchestratorInstance.informerManager.Listen()
+}
+
+// Since informerManager's sharedInformerFactory is started with no resync period, it never syncs the
+// existing cluster objects to its Store when it's started.
+// pvcAdded provides no additional handling but it ensures that existing PVCs in the cluster
+// gets added to sharedInformerFactory's Store before it's started. Then using informerManager's
```

Comment:
So if we don't add this callback method, PVCLister cannot find (`Get()` & `List()`) the PVCs that exist in the k8s cluster before `informerManager` is started.
This is particularly useful across container restarts and ensures that maps are fully initialized with the current PVC info.

Reply:
What happens to the store if a PVC is deleted? Does it clear from the store as well? Curious to know since you have an empty impl for PVC add event and not for delete event.

Reply:
If PVC gets deleted, the event will come after the `sharedInformerFactory` has been started. In that case, kubernetes takes care of deleting it from the Store.
The reason we added this empty implementation is because I observed that existing cluster PVCs does not get synced to the Store, and PVCLister fails to find those PVCs.